tools/wake-format/emitter.cpp

```diff
@@ -1425,59 +1426,84 @@ wcl::doc Emitter::walk_literal(ctx_t ctx, CSTElement node) {
   MEMO(ctx, node);
   FMT_ASSERT(node.id() == CST_LITERAL, node, "Expected CST_LITERAL");
 
+  std::string::size_type prefix_size = 0;
+
+  if (node.firstChildElement().id() == TOKEN_MSTR_BEGIN ||
+      node.firstChildElement().id() == TOKEN_MSTR_RESUME ||
+      node.firstChildElement().id() == TOKEN_LSTR_BEGIN ||
+      node.firstChildElement().id() == TOKEN_LSTR_RESUME) {
+    prefix_size = MultiLineStringIndentationFSM::analyze(node);
+  }
+
   // clang-format off
-  auto mstr_fmt = fmt()
-    .match(
-      pred(TOKEN_MSTR_BEGIN, fmt().token(TOKEN_MSTR_BEGIN))
-     .pred(TOKEN_MSTR_RESUME, fmt().token(TOKEN_MSTR_RESUME))
-     // No otherwise, this should fail if neither are true
-    )
-    .fmt_while(
-      {TOKEN_NL, TOKEN_WS, TOKEN_MSTR_CONTINUE},
-      fmt().match(
-        pred(TOKEN_WS, fmt().token(TOKEN_WS))
-       .pred(TOKEN_NL, fmt().token(TOKEN_NL))
-       .pred(TOKEN_MSTR_CONTINUE, fmt().token(TOKEN_MSTR_CONTINUE))
-      ))
-    .match(
-      pred(TOKEN_MSTR_PAUSE, fmt().token(TOKEN_MSTR_PAUSE))
-     .pred(TOKEN_MSTR_END, fmt().token(TOKEN_MSTR_END))
-     // No otherwise, this should fail if neither are true
-    );
 
-  auto lstr_fmt = fmt()
+  // Insert the proper amount of spaces to correctly indent the line relative to base identation
+  auto inset_line = fmt().escape([prefix_size](wcl::doc_builder& builder, ctx_t ctx, CSTElement& node){
+           FMT_ASSERT(node.id() == TOKEN_WS, node, "Expected <TOKEN_WS>, Saw <" + std::string(symbolName(node.id())) + ">");
+           for (size_t i = prefix_size; i < node.fragment().segment().size(); i++) {
+             builder.append(" ");
+           }
+           node.nextSiblingElement();
+  });
```

Comment:
Spoke off line about this but I think there's a slight issue if someone types an actual tab character into a string here instead of using \t. As is it will be replaced with a space. That would be ok IMO if it were documented and matched how the runtime behaved but I don't think it does.

Reply:
Okay it is updated to do the replace as it should but I don't think wake in general correctly handles this case